controlplane/kubeadm/internal/machinefilters/machine_filters.go

```diff
@@ -154,16 +152,15 @@ func IsReady() Func {
 
 // ShouldRolloutAfter returns a filter to find all machines
 // that should rollout after the given time.
-func ShouldRolloutAfter(t *metav1.Time) Func {
-	now := metav1.Now()
+func ShouldRolloutAfter(now, upgradeAfter *metav1.Time) Func {
 	return func(machine *clusterv1.Machine) bool {
 		if machine == nil {
 			return false
 		}
-		if t == nil || !t.Before(&now) {
+		if upgradeAfter == nil || !upgradeAfter.Before(now) {
```

Comment:
It looks like at least as of [v1.15 metav1.Time](https://github.com/kubernetes/apimachinery/blob/release-1.15/pkg/apis/meta/v1/time.go) can handle the nil check for us, so we should be able to remove it, it also might help to invert the check:
```suggestion
		if now.Before(upgradeAfter) {
```

Reply:
Although if we are going that route, we can collapse it down with the check below as well:
```go
		return now.Before(upgradeAfter) && machine.CreationTimestamp.Before(upgradeAfter)
```